src/components/Wizard/WizardContainer/Languages/Languages.js

```diff
@@ -0,0 +1,361 @@
+		});
+	};
+	onSuggestionsFetchRequested = ({ value }) => {
+		this.setState({
+			suggestions: getSuggestions(value),
+		});
+	};
+	onSuggestionsClearRequested = () => {
+		this.setState({
+			suggestions: [],
+		});
+	};
+	addSelectedTag = (name) => {
+		if (this.state.selectedTags.length < 10) {
+			let popularTags = [...this.state.popularTags];
+			let selectedTags = [...this.state.selectedTags];
+			let index = popularTags.indexOf(name);
+			popularTags.splice(index, 1);
+			selectedTags.push(name);
+			this.setState({ popularTags, selectedTags });
+		} else {
+			this.setState({ inputError: "* exceeded limit" });
+		}
+	};
+	removeSelectedTag = (name) => {
+		if (popularTags.includes(name)) {
+			let selectedTags = [...this.state.selectedTags];
+			let popularTags = [...this.state.popularTags];
+			let index = selectedTags.indexOf(name);
+			selectedTags.splice(index, 1);
+			popularTags.push(name);
+			this.state.inputError = "";
+			this.setState({ popularTags, selectedTags });
+		} else {
+			let selectedTags = [...this.state.selectedTags];
+			let index = selectedTags.indexOf(name);
+			selectedTags.splice(index, 1);
+			this.setState({ selectedTags });
+		}
+	};
+	addInputTag = (event) => {
+		if (event.key === "Enter" && this.state.value) {
+			const blockedRegex = /[\[\]!@#$%^&*(),.?":{}|<>]/;
+
+			if (this.state.value.match(blockedRegex)) {
+				this.setState({ inputError: "* only string values" });
+			} else {
+				if (this.state.value.length < 30) {
+					let value = this.state.value;
+					let valueLowerCase = this.state.value.toLowerCase();
+					let selectedTags = [...this.state.selectedTags];
+					let popularTags = [...this.state.popularTags];
+
+					let popularTagsLowerCase = popularTags.map((lowerCase) => {
+						return lowerCase.toLowerCase();
+					});
+
+					let selectedTagsLowerCase = selectedTags.map((lowerCase) => {
+						return lowerCase.toLowerCase();
+					});
+
+					if (popularTagsLowerCase.includes(valueLowerCase)) {
+						let index = popularTagsLowerCase.indexOf(valueLowerCase);
+						selectedTags.push(popularTags[index]);
+						popularTags.splice(index, 1);
+						value = "";
+						this.setState({ selectedTags, popularTags, value });
+					} else {
+						if (!selectedTagsLowerCase.includes(valueLowerCase)) {
+							selectedTags.push(value);
+							value = "";
+							this.setState({ selectedTags, value });
+						} else {
+							return;
+						}
+					}
+				} else {
+					this.setState({ inputError: "* too long string" });
+				}
+			}
+		}
+	};
+	componentDidMount() {
+		if (this.props.containerState) {
+			this.setState({
+				popularTags: this.props.containerState.popularTags,
+				selectedTags: this.props.containerState.selectedTags,
+			});
+		}
+		this.setState({ popularTags });
+	}
+	validation = () => {
+		let inputError = "";
+
+		console.log("IM DOING VALIDATION");
+
+		if (this.state.selectedTags.length === 0) {
+			inputError = "* chose a language";
+		}
+
+		if (inputError) {
```

Comment:
Is there any reason why we are not just doing:
```javascript
if(this.state.selectedTags.length === 0) {
	this.setState({ * choose a language});
	return false;
}
return true;
```

Reply:
Yet again, you are absolutely right, i've used previous step validation where we had 3 inputs with different validation error messages, with only one input i will use your method.